pkg/controller/cluster/controller.go

```diff
@@ -23,33 +23,90 @@ import (
 
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
 
-	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"github.com/golang/glog"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util"
 )
 
 // +controller:group=cluster,version=v1alpha1,kind=Cluster,resource=clusters
 type ClusterControllerImpl struct {
 	builders.DefaultControllerFns
 
-	// lister indexes properties about Cluster
-	lister listers.ClusterLister
+	actuator Actuator
+
+	// clusterLister holds a lister that knows how to list Cluster from a cache
+	clusterLister listers.ClusterLister
+
+	kubernetesClientSet *kubernetes.Clientset
+	clientSet           clientset.Interface
+	clusterClient       v1alpha1.ClusterInterface
 }
 
 // Init initializes the controller and is called by the generated code
 // Register watches for additional resource types here.
-func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments) {
-	// Use the lister for indexing clusters labels
-	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters().Lister()
+func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments, actuator Actuator) {
+	cInformer := arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters()
```

Comment:
style: all the extra newlines in this function are adding questionable value, esp the instances of double new lines on 57 and 58, suggest reducing the number or removing all new lines.

Reply:
Fixed, thanks!